src/mod_zone_difficulty_scripts.cpp

```diff
@@ -229,13 +922,26 @@ class mod_zone_difficulty_unitscript : public UnitScript
                             int32 absorb = eff->GetAmount();
                             uint32 phaseMask = target->GetPhaseMask();
                             int matchingPhase = sZoneDifficulty->GetLowestMatchingPhase(mapId, phaseMask);
-                            if (sZoneDifficulty->ZoneDifficultyInfo[mapId][matchingPhase].Enabled && (matchingPhase != -1))
+                            int8 mode = sZoneDifficulty->NerfInfo[mapId][matchingPhase].Enabled;
+                            if (matchingPhase != -1)
                             {
-                                absorb = eff->GetAmount() * sZoneDifficulty->ZoneDifficultyInfo[mapId][matchingPhase].AbsorbNerfPct;
+                                Map* map = target->GetMap();
+                                if ((mode & MODE_NORMAL) == MODE_NORMAL)
```

Comment:
You use this 6 times in the file and the chekc for hardmode 7 times. If you want to clean it up and avoid this, pardon my french, ugliness I would suggest you create a helper to get mode. Something like

```cpp
bool GetNormalMode(int32 mode) { return (mode & MODE_NORMAL) == MODE_NORMAL; }
bool GetHardMode(int32 mode) { return (mode & MODE_HARD) == MODE_HARD; }
```
In a header file ofc.


```cpp
if (GetNormalMode(mode))
```